[KeyVault] READMEs for the tests (#6145)
Azure/azure-sdk-for-js: sdk/keyvault/keyvault-certificates/test/README.md

# Testing

> To run our tests, first install the dependencies by running `npm install` (or `rush install`) at the root of the source code of this library (keyvault-certificates).

Reviewer: Doesnt the user have to run rush on the root of the repo first?

Author: Not necessarily! If they decide to just test this library, they can use `npm install`, I believe.

This will run tests in a purely nodejs environment, and in a browser environment using [Karma](https://karma-runner.github.io/latest/index.html).
These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses.
You won't need to have any environment configuration, nor internet access, for these tests to run.

Keep in mind that if you modify the test files, the API shape of this library, or the underlying network calls, the recordings need to be re-generated.
You can read more in the section: [Regenerating recordings](#regenerating-recordings).

## Integration tests

Our integration tests will run against the live resources, which are determined by the environment variables you provide.
To run the integration tests, you will need to execute `npm run integration-test`.

The environment variables needed are:

- `AZURE_CLIENT_ID`: The Client ID of your Azure account.
- `AZURE_CLIENT_SECRET`: The secret of your Azure account.
- `AZURE_TENANT_ID`: The Tenant ID of your Azure account.
- `KEYVAULT_NAME`: The name of the Key Vault you want to run the tests against.

> Integration tests won't need updated recordings, and they also won't change previous recordings unless specified (see the section below).

Reviewer: I prefer links rather than "see the section below".
In the future if we move things around or we add more sections, then links would still apply, but the below/top references might not

## Regenerating recordings

If you modify the test files, the API shape of this library, or the underlying network calls, the recordings need to be re-generated before you run the unit tests.

> To re-generate the recordings, populate the environment variable `TEST_MODE` with the value `record` before running the integration tests. Once the integration tests finish,

Reviewer: populate -> set ?

```suggestion
To re-generate the recordings, set the environment variable `TEST_MODE` to `record` and then run the integration tests. Once the integration tests finish,
```

the recordings will be updated.

## The common recorder

Our tests use our library [test-utils-recorder](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/test-utils/recorder).

> For any other question or if you want to help us get better at testing, please continue reading the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).

Reviewer: ```suggestion
For any other question or if you want to help us get better at testing, please read the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).
```
